falcon/routing/compiled.py

```diff
@@ -206,7 +206,7 @@ def line(text, indent_offset=0):
                     #   /foo/{id}/bar
                     #   /foo/{name}/bar
                     #
-                    assert len(nodes) == 1
+                    assert len([node for node in nodes if node.is_var]) == 1
```

Comment:
This is a simple test, perhaps something more robust with order checking would be helpful.